dts/bindings/input/espressif,touch-sensor.yaml

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Zephyr input touch sensor parent node
+
+  This defines a group of touch sensors that can generate input events. Each touch
+  sensor is defined in a child node of the touch-sensor node and defines a specific key
+  code.
+
+  For example:
+
+  #include <zephyr/dt-bindings/input/input-event-codes.h>
+  #include <zephyr/dt-bindings/input/espressif-input.h>
+
+  / {
+         touch_sensors {
+                 compatible = "espressif,touch-sensor";
+                 status = "okay";
+                 debounce-interval-ms = <30>;
+                 href-volt = <ESP32_TOUCH_HREF_VOLT_2V7>;
+                 lref-volt = <ESP32_TOUCH_LREF_VOLT_0V5>;
+                 href-atten = <ESP32_TOUCH_HVOLT_ATTEN_1V>;
+                 filter-mode = <ESP32_TOUCH_FILTER_MODE_IIR_16>;
+                 filter-debounce-cnt = <1>;
+                 filter-noise-thr = <ESP32_TOUCH_FILTER_NOISE_THR_4_8TH>;
+                 filter-jitter-step = <4>;
+                 filter-smooth-level = <ESP32_TOUCH_FILTER_SMOOTH_MODE_IIR_2>;
+
+                 touch_sensor_0 {
+                         channel_num = <1>;
+                         channel_sens = <20>;
+                         zephyr,code = <INPUT_KEY_0>;
+                 };
+         };
+  };
+
+
+compatible: "espressif,touch-sensor"
+
+include: base.yaml
+
+properties:
+  debounce-interval-ms:
+    type: int
+    default: 30
+    description: Debouncing interval time in milliseconds.
+
+  href-volt:
+    type: int
+    default: 3
+    description: |
+       Touch sensor high reference voltage.
+       If not specified defaults to ESP32_TOUCH_HREF_VOLT_2V7.
+
+  lref-volt:
+    type: int
+    default: 0
+    description: |
+       Touch sensor low reference voltage.
+       If not specified defaults to ESP32_TOUCH_LREF_VOLT_0V5.
```

Comment:
please, use values in SI units and appropriate property suffixes, you can later make use of enum indexes to make code more efficient.